Assets/Scripts/DayNightCycle/DayLengthController.cs

```diff
@@ -0,0 +1,56 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+using Assets.Scripts.Services;
+using Assets.Scripts.Settings;
+using UnityEngine.SceneManagement;
+namespace Assets.Scripts.DayNightCycle
+{
+    public class DayLengthController : MonoBehaviour
+    {
+        public Text Label;
+        public InputField DayLengthInput;
+        DayAndNightControl dayNightCycle;
+
+        void Start()
+        {
+            dayNightCycle = GameObject.Find("Day and Night Controller").GetComponent<DayAndNightControl>();
```

Comment:
Let's not use hardcoded strings to find something. Where are the references?

Reply:
I don't know if I can reference the Day and Night Controller because it is added to every scene, so basically the only way to find it is when the UI component is enabled in a scene, to then find the local Day and Night Controller game object.

Also to note DayAndNightControl.cs itself is about 0.3% of CPU ms usage so doesn't significantly affect performance. Lighting and Rendering is by far the larger culprit of performance drops thus far I think(from unity profiler)